griptape/structures/pipeline.py

```diff
@@ -53,8 +53,8 @@ def insert_task(self, parent_task: BaseTask, task: BaseTask) -> BaseTask:
     def try_run(self, *args) -> Pipeline:
         self.__run_from_task(self.input_task)
 
-        if self.conversation_memory and self.output is not None:
-            run = Run(input=self.input_task.input, output=self.output)
+        if self.conversation_memory and self.output_task.output is not None:
```

Comment:
Is there any desire to consolidate the conversation memory logic of all the structures (L56-59 here, L117-120 in workflow)? Noticed the same code when tinkering with a custom Agent locally and now this PR is surfacing it again.

Reply:
Sure thing, it would consolidate nicely into `Structure.after_run`. Would you mind making that PR?

Reply:
Actually since there's no change in behavior I can just cram it into this one.